packages/requisitions/src/ResponseRequisition/api/hooks.ts

```diff
@@ -133,3 +138,46 @@ export const useSaveResponseLines = () => {
       ]),
   });
 };
+
+const useOpenInNewTab = () => {
+  const navigate = useNavigate();
+  const { origin } = window.location;
+  return (url: string) => {
+    const to = `${origin}${url}`;
+    console.log(origin);
+    console.log(to);
+    const win = window.open(to, '_blank');
+    if (win) {
+      win.focus();
+    } else {
+      navigate(to);
+    }
+  };
+};
+
+export const useCreateOutboundFromResponse = () => {
+  const { error, warning } = useNotification();
+  const t = useTranslation('distribution');
+  const openInNewTab = useOpenInNewTab();
+  const { id } = useResponseFields('id');
+  const api = useResponseApi();
+  return useMutation(() => api.createOutboundFromResponse(id), {
+    onSuccess: (invoiceNumber: number) => {
+      openInNewTab(
+        RouteBuilder.create(AppRoute.Distribution)
+          .addPart(AppRoute.OutboundShipment)
+          .addPart(String(invoiceNumber))
+          .build()
+      );
+    },
+    onError: e => {
```

Comment:
I believe you could type this a bit more specifically to be the `CreateRequisitionShipmentError` type. In general I think a relatively good way to handle errors.

Reply:
very nice - yeah. that's a neat way to catch and respond